src/EnergyPlus/AirLoopHVACDOAS.cc

```diff
@@ -961,6 +954,31 @@ namespace AirLoopHVACDOAS {
         }
         state.dataSize->CurSysNum = state.dataHVACGlobal->NumPrimaryAirSys + this->m_AirLoopDOASNum + 1;
         state.dataSize->CurOASysNum = this->m_OASystemNum;
+
+        // Add fan heat to cooling coil capacity
+        if (this->FanHeatAdded) {
+            Real64 deltaP;
+            Real64 motEff;
+            Real64 totEff;
+            Real64 motInAirFrac;
+            Real64 fanShaftPow;
+            Real64 motInPower;
+            bool fanCompModel;
+            Real64 designHeatGain = 0.0;
+            Real64 airVolFlow = sizingMassFlow / state.dataEnvrn->StdRhoAir;
+            if (this->m_FanTypeNum == SimAirServingZones::CompType::Fan_System_Object) {
+                state.dataHVACFan->fanObjs[this->m_FanIndex]->FanInputsForDesignHeatGain(state, deltaP, motEff, totEff, motInAirFrac);
+                Real64 fanPowerTot = (airVolFlow * deltaP) / totEff;
+                designHeatGain = motEff * fanPowerTot + (fanPowerTot - motEff * fanPowerTot) * motInAirFrac;
+            } else if (this->m_FanTypeNum == SimAirServingZones::CompType::Fan_ComponentModel) {
+                Fans::FanInputsForDesHeatGain(state, this->m_FanIndex, deltaP, motEff, totEff, motInAirFrac, fanShaftPow, motInPower, fanCompModel);
+                designHeatGain = fanShaftPow + (motInPower - fanShaftPow) * motInAirFrac;
+            }
+            // Calculate temperature diff
+            Real64 CpAir = Psychrometrics::PsyCpAirFnW(state.dataLoopNodes->Node(this->m_FanInletNodeNum).HumRat);
+            Real64 fanDeltaT = designHeatGain / (sizingMassFlow * CpAir);
+            this->PrecoolTemp -= fanDeltaT;
+        }
```

Comment:
Add fan heat

Reply:
I was not expecting this. To size a coil with fan heat all that is needed is to set the fan index, type and placement. Then the call to CoolingCapacitySizing will include the fan heat (if that section of code is corrected) in coil sizing. Was this method investigated? Is there a reason that DOAS systems need to account for fan heat? The DOAS design air flow should be known when the DOAS coils are sized. If this section of code is filled in then the coils should size correctly. If not then I would like to know why.

    if (this->curOASysNum > 0) { // coil is in the OA stream
        // need to find fan type in OA system
    } else {
        switch (this->primaryAirSystem(this->curSysNum).supFanModelType) {

Reply:
@lgu1234 please provide a reply here and we'll work on pushing this along the review train.

Reply:
@rraustad The mass flow is given. I tried to fill in the section code in CoolingCapacitySizing by calling CoolingCapacitySizer::size. You told me that before. For some reason, I was unable to connect the function size in AirLoopHVACDOAS. In other words, I was unable to set "auto coolingCoilSize &CoolingCapacitySizing". That was why I use the same functions in your this->calcFanDesHeatGain(DesVolFlow). Therefore, same functions are called to get fan heat. The difference is I set up locally.

Reply:
You don't call CoolingCapacitySizer from AirLoopHVACDOAS, the cooling coil calls that function. What you do here is set OutsideAirSys().supFanModelType and OutsideAirSys().supFanLocation and let the sizing function calculate the fan heat gain, delta T, and adjustment to coil capacity. You are adding this for DOAS systems but there can also be coils in the OutsideAirSystem so be generic to allow coils in OA systems to also use this code. Then all you need to do is fill in "// need to find fan type in OA system" with similar code to what is below here.

    if (this->curOASysNum > 0) { // coil is in the OA stream
        // need to find fan type in OA system
        } else {
            switch (this->primaryAirSystem(this->curSysNum).supFanModelType) {
            case DataAirSystems::StructArrayLegacyFanModels: {
                FanCoolLoad = this->calcFanDesHeatGain(DesVolFlow);
                break;
            }
            case DataAirSystems::ObjectVectorOOFanSystemModel: {
                FanCoolLoad = this->calcFanDesHeatGain(DesVolFlow);
                break;
            }
            default:
                // do nothing
                break;
            } // end switch

Then you should not have to adjust the DeltaT here:

                } else if (this->curOASysNum > 0 && this->outsideAirSys(this->curOASysNum).AirLoopDOASNum > -1) {
                    auto &thisAirloopDOAS = this->airloopDOAS[this->outsideAirSys(this->curOASysNum).AirLoopDOASNum];
                    if (thisAirloopDOAS.m_FanIndex > -1 && thisAirloopDOAS.FanBlowTroughFlag &&
                        thisAirloopDOAS.m_FanTypeNum == SimAirServingZones::CompType::Fan_System_Object) {
                        int FanIndex = thisAirloopDOAS.m_FanIndex;
                        Real64 DeltaT = state.dataHVACFan->fanObjs[FanIndex]->getFanDesignTemperatureRise(state);
                        CoilInTemp += DeltaT;
                    }

Reply:
@rraustad I debug the code and found that it never reach your section. Instead, code go through Line 296.

Reply:
@lgu1234 that's because of the private "else if" for AirLoopHVACDOAS. That section of code should be merged into the else section. All that's needed is the inlet and outlet conditions and DesVolFlow to be set properly. If you run a file that has an OutdoorAirSystem it will go through the else section.

Reply:
@rraustad Actually, I found the code reach line 275 in CoolingCapacitySizing. Your section is not reached. Here is a section of code:

                    if (thisAirloopDOAS.m_FanIndex > -1 && thisAirloopDOAS.FanBlowTroughFlag &&
                        thisAirloopDOAS.m_FanTypeNum == SimAirServingZones::CompType::Fan_System_Object) {
                        int FanIndex = thisAirloopDOAS.m_FanIndex;
                        Real64 DeltaT = state.dataHVACFan->fanObjs[FanIndex]->getFanDesignTemperatureRise(state);
                        CoilInTemp += DeltaT;
                    }

I already added fan heat, if the fan is blowtrough. Therefore, I need to add one more scenario when a fan is placed before the cooling coil.

Reply:
OK, go ahead and use that else if block to size a coil for AirLoopHVACDOAS. Too much would need to be done to avoid conflicts with FinalSysSizing in the else part of those blocks.